cmd/antrea-agent/agent.go

```diff
@@ -325,21 +326,18 @@ func run(o *Options) error {
 
 	// Initialize localPodInformer for NPLAgent, AntreaIPAMController,
 	// StretchedNetworkPolicyController, and secondary network controller.
-	var localPodInformer cache.SharedIndexInformer
-	if o.enableNodePortLocal || enableBridgingMode || enableMulticlusterNP || enableFlowExporter ||
-		features.DefaultFeatureGate.Enabled(features.SecondaryNetwork) ||
-		features.DefaultFeatureGate.Enabled(features.TrafficControl) {
+	localPodInformer := lazy.New[cache.SharedIndexInformer](func() cache.SharedIndexInformer {
```

Comment:
would it be equivalent to use `sync.OnceValue`?

```go
localPodInformerGet := sync.OnceValue(func() cache.SharedIndexInformer {
    // ...
    return coreinformers.NewFilteredPodInformer(...)
})

// ...
localPodInformerGet()
localPodInformerGet()
```

Reply:
Thanks for the suggestion, I didn't know this function but there are still two reasons why I feel the current implementation is more suitable:
1. It's more readable when we need to pass the "getter" to another function like why I need the interface:
https://github.com/antrea-io/antrea/blob/6831d5701c0bfeb1002c69d88ae619773f48e6f4/pkg/agent/controller/networkpolicy/networkpolicy_controller.go#L148C2-L148C13
2. It takes less code on caller side to know whether it's ever evaluated. Otherwise we will have to have another local varible to track it for latter running it.